### PR TITLE
[AMORO-2365] Fix get target snapshot NPE when commiting

### LINKED ISSUE #2365: [Bug]: Expired the snapshot that is planning in optimizing process
### What happened?

Currently, we create `TableOptimizingProcess` after the planner is ready:
image

When optimizing tasks are not committed, the snapshot with which it planned should not be expired:
image

However, when a table is planned for too long, the snapshot of the table may be expired during planning. Because the `TableOptimizingProcess` has not been generated at this time, `fetchOptimizingSnapshotTime` will return `Long.MAX_VALUE`.

Then when we commit, we'll occur NPE:
image

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

Construct a scenario as described in ‘what happened’.

### Relevant log output

```shell
NPE problem.
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by HuangFru**
## Why are the changes needed?

Close #2365.

## Brief change log

- Add 'PLANNING' status in `OptimizingProcess.Status' and set it as default.
- Generic the `TableOptimizingProcess` when the table is planning and update it after planned.
- Change judgment in 'SnapshotExpire' to not expire the snapshot that is used for planning.
- Change the init SQL to make 'optimizing_type' default null.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by HuangFru** [comment]
**When the table is planning:**
image

image

**After planning:**
image

image

**When the table is planning:**
image
image

**Restart the AMS:**
image

**When the table is planning I set `self-optimizing.enabled` to 'fasle':**
image
image

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingProcess.java [hunk 1]
```diff
@@ -6,6 +6,8 @@ public interface OptimizingProcess {
 
   void close();
 
+  void failed(String failedReason, long endTime);
```
[the remark this answers is not in the document]
Comment (HuangFru) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingProcess.java]: got it.

**Comment by HuangFru** [comment]
> I left some comments, please take a look @HuangFru

Fixed.

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java [hunk 2]
```diff
@@ -308,19 +315,20 @@ SchedulingPolicy getSchedulingPolicy() {
...
     private final Map<OptimizingTaskId, TaskRuntime> taskMap = Maps.newHashMap();
     private final Queue<TaskRuntime> taskQueue = new LinkedList<>();
     private final Lock lock = new ReentrantLock();
-    private volatile Status status = OptimizingProcess.Status.RUNNING;
+    private volatile Status status = OptimizingProcess.Status.PLANNING;
     private volatile String failedReason;
     private long endTime = ArcticServiceConstants.INVALID_TIME;
     private Map<String, Long> fromSequence = Maps.newHashMap();
     private Map<String, Long> toSequence = Maps.newHashMap();
+    private final OptimizingPlanner planner;
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java:331]: I think the main issue here is that the planner, as a field of the optimizing process, is not being persisted, which leads to the process having to be closed after restarting AMS. 
My suggestion is not to add this field `planner`, but instead to construct a new planner when calling `planTasks()` and pass it as a parameter, and the `targetSnapshotId` and `targetChangeSnapshotId` will be set when planning tasks.

**Pull request closed without merging**